resources/mysql2_gem.rb

```diff
@@ -0,0 +1,25 @@
+provides :mysql2_gem
+resource_name :mysql2_gem
+
+property :compile_time, [true, false], default: false
+
+default_action :install
+
+action :install do
+  include_recipe 'mariadb::devel'
+  build_essential 'install_packages'
+  gem_package 'mysql2'
+end
+
+action :remove do
+  gem_package 'mysql2' do
+    action :remove
+  end
+end
+
+def after_created
+  return unless compile_time
+  Array(action).each do |action|
+    run_action(action)
+  end
+end
```

Comment:
It could be a recipe, I agree. The only problem with recipe would be to manage when to install packages (compile or converge phaze).

Client recipe installs devel package only if attribute for that is set. So I would need to assign or override node attribute from custom resource. And I'm not sure if doing that is correct approach.